# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

The decision for me was between Lane 2 and 4. I chose Lane 4 over Lane 2 because it asks a narrower, more directly testable question: are visible pages under-capturing clicks relative to what's normal for their position? My own EDA (grouping CTR by position tier) gave me real evidence that "normal" CTR varies a lot by tier, which is exactly the kind of gap this lane is built to catch.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Search question**: Which pages are underperforming on CTR relative to other pages in the same position tier, and therefore may warrant a title/meta review?
**Unit of analysis**: One row in my analysis represents a page: CTR, impressions, and position tier are all measured at the page level.
**Output**: A ranked table of pages, where each row shows the page's CTR, impressions, position tier, and a flag/reason indicating it's underperforming relative to its own tier, so a reviewer can prioritize which titles to rework first.
**Decision / action**: content review or a SEO person responsible for CTR of most impression pages. They evaluate the CTR and the impressions and if the CTR does not match with the impressions its getting then the flaw is in the title and they rewrite that title so it appeals more masses and they actually click on that page.

**Cost of a wrong call:**

**False positive**: wastes the reviewer's time on a page that didn't actually need a fix, time that could've gone to a page that did.
**False negative**: the page stays unaddressed, and every day it's not fixed, more people see it in search results and don't click. That's compounding, ongoing lost traffic, not a one-time cost.
**Verdict**: false negatives are worse, since the cost compounds over time instead of being a single wasted review.

Why data/ML, not just a rule: A single fixed CTR threshold doesn't work because "normal" CTR is different for every position tier, a low CTR is expected for a deep-tier page but is a real problem for a top_3 page. A flat if-statement can't make that per-tier, context-dependent comparison; each page needs to be judged against others in its own tier.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import pandas as pd
df = pd.read_csv("/content/FlyRank-Intern/data/raw/content_refresh_anonymized.csv")
df = df[df["avg_position"] > 0]

df.groupby("position_tier")["ctr"].agg(["mean", "median", "count"])


,mean,median,count
position_tier,,,
deep,0.150212,0.00,1319
page_1,0.652467,0.16,11814
page_3_5,0.222484,0.03,7242
striking,0.323239,0.11,7304
top_3,2.764453,0.00,1116


The mean and median diverge sharply in several tiers, especially top_3 (mean 2.76 vs. median 0.00) meaning a few outlier pages are skewing the average, and "better tier = better CTR" isn't a clean pattern. This divergence is itself evidence that no single fixed CTR threshold would treat pages fairly across tiers, each page needs to be compared to its own tier's typical performance.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My analysis can claim that CTR varies significantly by position tier, and that this variation points toward which page titles might need rework — but it's decision support, not a fix. I can't claim that rewriting a title will actually fix the CTR of that page. The mean and median are messy enough that I can't treat them as a settled rule. And I'm not making any claim about how Google's algorithm works or how search ranking actually functions, everything here comes purely from observing this dataset. I also can't claim that being flagged as low-CTR-for-its-tier means the title is the actual cause of underperformance. The flag only marks a candidate worth a closer look, not a confirmed diagnosis.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.